resources/schema/Rule_Type.md

```diff
@@ -0,0 +1,228 @@
+## Dataset Contents Check against Define XML and Library Metadata
+
+#### Columns
+
+Columns are the columns within the original dataset
+
+#### Rule Macro
+
+Attach define xml metadata at variable level and library metadata at variable level
+
+## Dataset Metadata Check
+
+#### Columns
+
+- `dataset_label`
+- `dataset_location`
+- `dataset_name`
+- `dataset_size`
+
+#### Example
+
+```yaml
+- name: dataset_name
+  operator: longer_than
+  value: 6
+```
+
+## Dataset Metadata Check against Define XML
+
+#### Columns
+
+- `dataset_label`
+- `dataset_location`
+- `dataset_name`
+- `dataset_size`
+
+#### Rule Macro
+
+Allows comparing content metadata to define xml metadata of the same name.
+
+#### Example
+
+```yaml
+- name: dataset_label
+  operator: not_equal_to
+```
+
+## Define Item Group Metadata Check
+
+#### Columns
+
+- `define_dataset_name`
+- `define_dataset_label`
+- `define_dataset_location`
+- `define_dataset_class`
+- `define_dataset_structure`
+- `define_dataset_is_non_standard`
+- `define_dataset_variables`
+
+## Define Item Metadata Check
+
+#### Columns
+
+- `define_variable_name`
+- `define_variable_label`
+- `define_variable_data_type`
+- `define_variable_role`
+- `define_variable_size`
+- `define_variable_ccode`
+- `define_variable_format`
+- `define_variable_allowed_terms`
+- `define_variable_origin_type`
+- `define_variable_is_collected`
+- `define_variable_has_no_data`
+- `define_variable_order_number`
+- `define_variable_has_codelist`
+- `define_variable_codelist_coded_values`
+
+#### Rule Macro
```

Comment:
What is the rule macro?

Reply:
This is referring to the "SPECIAL CASES FOR RULE TYPES". I'm not sure what to call this. Rule preprocessing? I'm also not sure if all of these are correct since this part is a bit unclear to me.
 https://github.com/cdisc-org/cdisc-rules-engine/blob/bca332136648d22917c34ec93051c4f731ac0c2c/cdisc_rules_engine/rules_engine.py#L194

Reply:
I think we can call them rule macros, but I'm not sure if these need to be documented since the users cannot reference this data.